### PR TITLE
Release v0.10.0-alpha

### PR DESCRIPTION
**Pull request by tcharding**
In preparation for doing an `alpha` release of the new `primitives` crate bump the version to `0.10.0-alpha` and add a changelog entry.

Let's Go!

### CODE REVIEW

**Comment by tcharding** [comment]
@clarkmoody could you please concept ack/nack?

**Comment by clarkmoody** [comment]
ACK from me 👍

**Comment by tcharding** [comment]
Legend.

**Comment by apoelstra** [comment]
@tcharding is this good to go? Like, if we merge this can we do the release?

**Comment by tcharding** [comment]
Yep we are good to go, merge and release please.

**Comment by apoelstra** [comment]
@tcharding could you rebase this to remove the overlapping code from #118? It looks like this isn't causing merge conflicts but I think that's git being too smart for its own good.

**Review by apoelstra — APPROVED** [review]
ACK 202df4f19b61b2b8304773156009efe4feec9c1a

**Comment by apoelstra** [comment]
Oops, I should've done `cargo publish --dry-run` before merging.

@tcharding crates.io says we need to remove one of the keywords from Cargo.toml (have 5, max 6). Added in #98.

**Comment by tcharding** [comment]
Done, #123

**Pull request merged**
